hl7apy/core.py

```diff
@@ -29,6 +29,16 @@
 from itertools import takewhile
 import importlib
 
+try:
+    basestring
+except NameError:
+    basestring = str
+
+try:
+    xrange
+except NameError:
+    xrange = range
+
```

Comment:
Ok for the try/except approach but I don't like so much the instruction which doesn't do anything but raising the exception. I'd rather use a solution where both the try and except parts assign their version of str/range to another variable and then use it in the rest of the code. I think this way the intent is more clear.
For example:

    try:
        str_type = basestring
    except NameError:
        str_type = str

and the same for xrange

    try:
        range_fun = xrange
    except NameError:
        range_fun = range